drivers/adc/adc_stm32.c

```diff
@@ -248,6 +248,25 @@ static int check_buffer_size(const struct adc_sequence *sequence,
 	return 0;
 }
 
+static uint32_t adc_stm32_getChannelByIndx(const uint8_t index) {
+	uint32_t channel = __LL_ADC_DECIMAL_NB_TO_CHANNEL(index);
+	/*
+	 * the stm32f4xx offers internal channels
+	 * they could be used through dts flags
+	 * for now only the channel 17 (VRef internal)
+	 * is implemented
+	 */
+#if defined(CONFIG_SOC_SERIES_STM32F4X)
+#if (DT_PROP(DT_NODELABEL(adc1), ch17_internal) == 1)
```

Comment:
Is that only valid for adc1 ?
Also, one problem raised here is that this piece of code will be applied whatever the instance in use, as long as ch17_internal is defined for adc1.

I'd advice to store the information about ch17_internal in a device specific struct (adc_stm32_cfg),
and check for the info in that structure in run time. So you ensure that channel setting is done only on the expected instance (be it adc1, adc2, adcx or any combination)

Reply:
thx for the comment
I just dived into DT and its macros. So I am still beginner - if you have an better idea pls let me know. 
Storing the internal/external channel info must be handled at one instance. I used macros that are fixed at compile time. The adc_stm32_cfg is a good idea, because its constant and therefore the compiler can resolve this const expr into static code. So that should not cost any performance (because the channel setup is done for every conversation). An other topic is in general the adc API, what translates the sequence into the device specific adc sequence for EVERY adc_read call. IMHO that is a bad design ;)